inspirehep/modules/theme/templates/inspirehep_theme/accounts/login.html

```diff
@@ -33,7 +33,7 @@
 <div class="col-xs-12 col-sm-10 col-md-8 col-lg-6 col-centered">
   <div class="well">
     <br/>
-    {% set next = request.url if request.endpoint != 'security.logout' and request.endpoint != 'security.login' else config.SERVER_NAME %}
+    {% set next = request.url if request.endpoint != 'security.logout' and request.endpoint != 'security.login' else request.args.get('next', '') %}
```

Comment:
what about fixing a default macro in https://github.com/inveniosoftware/invenio-oauthclient/blob/master/invenio_oauthclient/templates/invenio_oauthclient/_macros.html#L54 ?